src/avutil_stubs.c

```diff
@@ -40,3 +42,174 @@ CAMLprim value caml_avutil_bits_per_pixel(value pixel)
+  AV_CH_LAYOUT_5POINT1,
+  AV_CH_LAYOUT_5POINT0_BACK,
+  AV_CH_LAYOUT_5POINT1_BACK,
+  AV_CH_LAYOUT_6POINT0,
+  AV_CH_LAYOUT_6POINT0_FRONT,
+  AV_CH_LAYOUT_HEXAGONAL,
+  AV_CH_LAYOUT_6POINT1,
+  AV_CH_LAYOUT_6POINT1_BACK,
+  AV_CH_LAYOUT_6POINT1_FRONT,
+  AV_CH_LAYOUT_7POINT0,
+  AV_CH_LAYOUT_7POINT0_FRONT,
+  AV_CH_LAYOUT_7POINT1,
+  AV_CH_LAYOUT_7POINT1_WIDE,
+  AV_CH_LAYOUT_7POINT1_WIDE_BACK,
+  AV_CH_LAYOUT_OCTAGONAL,
+  AV_CH_LAYOUT_HEXADECAGONAL,
+  AV_CH_LAYOUT_STEREO_DOWNMIX
+};
+
+uint64_t ChannelLayout_val(value v)
+{
+  return CHANNEL_LAYOUTS[Int_val(v)];
+}
+
+value Val_channelLayout(uint64_t cl)
+{
+  for (int i = 0; i < CHANNEL_LAYOUTS_LEN; i++) {
+    if (cl == CHANNEL_LAYOUTS[i])
+      return Val_int(i);
+  }
+  printf("error in channel layout : %llu\n", cl);
+  return Val_int(0);
+}
+
+
+/**** Sample format ****/
+
+#define SAMPLE_FORMATS_LEN 10
+static const enum AVSampleFormat SAMPLE_FORMATS[SAMPLE_FORMATS_LEN] = {
+  AV_SAMPLE_FMT_U8,
+  AV_SAMPLE_FMT_S16,
+  AV_SAMPLE_FMT_S32,
+  AV_SAMPLE_FMT_FLT,
+  AV_SAMPLE_FMT_DBL,
+  AV_SAMPLE_FMT_U8P,
+  AV_SAMPLE_FMT_S16P,
+  AV_SAMPLE_FMT_S32P,
+  AV_SAMPLE_FMT_FLTP,
+  AV_SAMPLE_FMT_DBLP
+};
+
+enum AVSampleFormat SampleFormat_val(value v)
+{
+  return SAMPLE_FORMATS[Int_val(v)];
+}
+
+enum AVSampleFormat AVSampleFormat_of_Sample_format(int i)
+{
+  return SAMPLE_FORMATS[i];
+}
+
+value Val_sampleFormat(enum AVSampleFormat sf)
+{
+  for (int i = 0; i < SAMPLE_FORMATS_LEN; i++) {
+    if (sf == SAMPLE_FORMATS[i])
+      return Val_int(i);
+  }
+  printf("error in sample format : %d\n", sf);
+  return Val_int(0);
+}
+
+static const enum caml_ba_kind BIGARRAY_KINDS[SAMPLE_FORMATS_LEN] = {
+  CAML_BA_UINT8,
+  CAML_BA_SINT16,	
+  CAML_BA_INT32,	
+  CAML_BA_FLOAT32,
+  CAML_BA_FLOAT64,
+  CAML_BA_UINT8,
+  CAML_BA_SINT16,
+  CAML_BA_INT32,	
+  CAML_BA_FLOAT32,
+  CAML_BA_FLOAT64
+};
+
+enum caml_ba_kind bigarray_kind_of_Sample_format(int sf)
+{
+  return BIGARRAY_KINDS[sf];
+}
+
+enum caml_ba_kind bigarray_kind_of_Sample_format_val(value val)
+{
+  return BIGARRAY_KINDS[Int_val(val)];
+}
+
+enum caml_ba_kind bigarray_kind_of_AVSampleFormat(enum AVSampleFormat sf)
+{
+  for (int i = 0; i < SAMPLE_FORMATS_LEN; i++)
+    {
+      if (sf == SAMPLE_FORMATS[i])
+	return BIGARRAY_KINDS[i];
+    }
+  return CAML_BA_KIND_MASK;
+}
+
+CAMLprim value ocaml_avutil_get_sample_fmt_name(value _sample_fmt) {
+  CAMLparam1(_sample_fmt);
+  CAMLlocal1(ans);
+
+  const char *name = av_get_sample_fmt_name(SampleFormat_val(_sample_fmt));
+  ans = caml_copy_string(name);
+
+  CAMLreturn(ans);
+}
+
+
+/***** AVFrame *****/
+
+static void finalize_frame(value v)
+{
+  struct AVFrame *frame = Frame_val(v);
+  av_frame_free(&frame);
+}
+
+static struct custom_operations frame_ops =
+  {
+    "ocaml_avframe",
+    finalize_frame,
+    custom_compare_default,
+    custom_hash_default,
+    custom_serialize_default,
+    custom_deserialize_default
+  };
+
+value value_of_frame(AVFrame *frame)
```

Comment:
Think you also need to properly wrap this using at least `CAMLlocal` for the local `value` variables.